functions/grapher/[slug].ts

```diff
@@ -146,7 +102,7 @@ async function handleHtmlPageRequest(
     const grapherPageResp = await env.ASSETS.fetch(url, { redirect: "manual" })
 
     if (grapherPageResp.status === 404) {
-        throw new StatusError(404)
+        return handlePageNotFound(env, grapherPageResp)
```

Comment:
One thing I noticed while tinkering with the code, the `For local testing` commented-out code block above here could be updated to call `fetch` from `env.ASSETS`

Reply:
I think a comment here would be useful that mentions that by passing in the grapherPageResp as the fallback we get a 404 status code but also an HTML 404 page. Otherwise this subtle detail might be lost in bigger refactorings in the future.

Reply:
Added a comment about `grapherPageResp`.

> commented-out code block above here could be updated to call `fetch` from `env.ASSETS`

Even if it works, I think it makes sense to leave it to a normal fetch, since we are not fetching from the assets (semantically), we are fetching from the internet.